From 368efebe542ff55445b9648f46ef0244f9744ca6 Mon Sep 17 00:00:00 2001
From: Roy Pledge <roy.pledge@nxp.com>
Date: Tue, 5 Sep 2017 16:47:49 -0400
Subject: [PATCH 189/767] drivers/staging/fsl_qbman: Fix SWP Mask in Error
 Handling

If an ECC error occurs QBMan stores data in a register for
debug purposes. If the memory is software portal memory then
15 bits are used. This patch adjusts the mask to be correct.

Signed-off-by: Roy Pledge <roy.pledge@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl_qbman/qman_config.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_qbman/qman_config.c b/drivers/staging/fsl_qbman/qman_config.c
index 9bb1e11a0a33..529f840117ea 100644
--- a/drivers/staging/fsl_qbman/qman_config.c
+++ b/drivers/staging/fsl_qbman/qman_config.c
@@ -254,7 +254,7 @@ static const struct qman_error_info_mdata error_mdata[] = {
 	QMAN_ERR_MDATA(0x01FF, 72, "WQ context memory"),
 	QMAN_ERR_MDATA(0x00FF, 240, "CGR memory"),
 	QMAN_ERR_MDATA(0x00FF, 302, "Internal Order Restoration List memory"),
-	QMAN_ERR_MDATA(0x01FF, 256, "SW portal ring memory"),
+	QMAN_ERR_MDATA(0x7FFF, 256, "SW portal ring memory"),
 	QMAN_ERR_MDATA(0x07FF, 181, "CEETM class queue descriptor memory"),
 	QMAN_ERR_MDATA(0x0FFF, 140, "CEETM extended SFDR memory"),
 	QMAN_ERR_MDATA(0x0FFF, 25, "CEETM logical FQ mapping memory"),
@@ -812,7 +812,7 @@ int qman_set_sdest(u16 channel, unsigned int cpu_idx)
 
 	if (!qman_have_ccsr())
 		return -ENODEV;
-	if ((qman_ip_rev & 0xFF00) == QMAN_REV31) {
+	if ((qman_ip_rev & 0xFFFF) == QMAN_REV31) {
 		/* LS1043A - only one L2 cache */
 		cpu_idx = 0;
 	}
-- 
2.17.0

